registration-processor/registration-processor-rest-client/src/main/java/io/mosip/registration/processor/rest/client/utils/RestApiClient.java

```diff
@@ -227,8 +228,11 @@ private HttpEntity<Object> setRequestHeader(Object requestType, MediaType mediaT
 				Iterator<String> iterator = httpHeader.keySet().iterator();
 				while (iterator.hasNext()) {
 					String key = iterator.next();
-					if (!(headers.containsKey("Content-Type") && key.equalsIgnoreCase("Content-Type")) && httpHeader.get(key).size()>0)
-						headers.add(key,httpHeader.get(key).get(0));
+					List<String> collection = httpHeader.get(key);
+					if ((collection != null && !collection.isEmpty())
+							&& !(headers.containsKey("Content-Type") && key.equalsIgnoreCase("Content-Type")))
```

Comment:
Can you not copy all the headers and simply replace the Content-Type or remove? Looks like this while loop can be avoided.

Reply:
Please ignor this commit it was commited by mistake please check updated commit with name Mos-19396 Latest Commit